windows/code/RNDeviceInfoCPP.h

```diff
@@ -35,16 +43,37 @@ namespace winrt::RNDeviceInfoCPP
       provider.Add(L"appName", getAppNameSync());
       provider.Add(L"brand", getBrandSync());
       provider.Add(L"model", getModelSync());
+      provider.Add(L"deviceType", getDeviceTypeSync());
     }
 
     bool isEmulatorHelper(std::string model)
     {
       return std::regex_match(model, std::regex(".*virtual.*", std::regex_constants::icase));
     }
 
-    bool isTabletHelper(std::string os)
-    {
-      return !std::regex_match(os, std::regex(".*windowsphone.*", std::regex_constants::icase));
+    // What is a tablet is a debateable topic in Windows, as some windows devices can dynamically switch back and forth.
+    // Also, see isTabletMode() instead of isTablet or deviceType.
+    // For right now, we generally are considered mobile as 'tablets', as opposed to desktop computers.
+    // More refinement should be applied into this area as neccesary. 
+    bool isTabletHelper()
+    {
+      std::string osName = winrt::to_string(Windows::Security::ExchangeActiveSyncProvisioning::EasClientDeviceInformation().OperatingSystem());
+      // AnalyticsInfo is considered flakey or unreliable by most sources to be used, but no recommended alternatives were discovered.
+      // DeviceForm potential but not inclusive results:
+      // [Mobile, Tablet, Television, Car, Watch, VirtualReality, Desktop, Unknown]
+      // DeviceFamily potential but not inclusive results:
+      // [Windows.Desktop, Windows.Mobile, Windows.Xbox, Windows.Holographic, Windows.Team, Windows.IoT]
+      auto deviceForm = AnalyticsInfo::DeviceForm();
+      auto deviceFamily = AnalyticsInfo::VersionInfo().DeviceFamily();
+      
+      bool isTabletByAnalytics = deviceForm == L"Tablet" || deviceForm == L"Mobile" || deviceFamily == L"Windows.Mobile";
+      bool isTabletByOsRegex = std::regex_match(osName, std::regex(".*windowsphone.*", std::regex_constants::icase));
```

Comment:
we don't support running on windows phone afaik (RNW requires RS3 IIRC) so I think you can remove the regex and EAS stuff.
FWIW AnalyticsInfo is not flakey or unreliable :) it might return values that don't match what you'd expect but that is by design (e.g. for quirked apps or SKUs), so we should always use it when determining whether to behave one way or another to be in line with other behavior that depends on the quirks.

Reply:
I didn't add windowsphone originally, I was only fixing the incorrect regex return value, but good to know and I will remove it completely.

Me calling it flakey and unrelabie, is exactly what I mean by not getting values you'd expect by design. But I will update the comment to say that instead.